src/main/java/core/jdbc/JdbcApi.java

```diff
@@ -0,0 +1,156 @@
+package core.jdbc;
+
+import core.util.ReflectionUtils;
+import org.springframework.core.LocalVariableTableParameterNameDiscoverer;
+import org.springframework.core.ParameterNameDiscoverer;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static core.jdbc.ConnectionManager.getConnection;
+import static core.util.ReflectionUtils.hasFieldMethod;
+import static core.util.StringUtil.upperFirstChar;
+
+// 너무 많은 일을 하고 있는거 같다.. convert를 따로 빼야할까? 어떻게 빼야할까...
+public class JdbcApi<T> {
+    private static final ParameterNameDiscoverer nameDiscoverer = new LocalVariableTableParameterNameDiscoverer();
+
+    public final Class<T> clazz;
+
+    public JdbcApi(Class<T> clazz) {
+        this.clazz = clazz;
+    }
+
+    public void execute(final String sql, final Object... values) {
+        try(Connection connection = getConnection()) {
+            PreparedStatement preparedStatement = prepareStatement(connection, sql, values);
+
+            preparedStatement.execute();
+        } catch (Exception e) {
+            throw new JdbcApiException("Fail to execute query : " + e.getMessage());
+        }
+    }
+
+    public List<T> findAll(final String sql, final Object... values) {
+        return executeQuery(sql, values);
+    }
+
+    public T findOne(final String sql, final Object... values) {
+        List<T> classes = executeQuery(sql, values);
```

Comment:
findAll 함수를 활용해도 되겠네요 :)
```suggestion
        List<T> results = findAll(sql, values);
```

Reply:
이러는 편이 로직 중복도 막을 수 있어서 더 좋겠네요 :)